<a href="https://colab.research.google.com/github/Ifhamsb/Ifhamsb/blob/main/Training_18th_July.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

mydata = pd.read_csv('/content/NHANES.csv')

mydata.head()

SEQN  ALQ101  ALQ110  ALQ130  SMQ020  RIAGENDR  RIDAGEYR  RIDRETH1  \
0  83732     1.0     NaN     1.0       1         1        62         3   
1  83733     1.0     NaN     6.0       1         1        53         3   
2  83734     1.0     NaN     NaN       1         1        78         3   
3  83735     2.0     1.0     1.0       2         2        56         3   
4  83736     2.0     1.0     1.0       2         2        42         4   

   DMDCITZN  DMDEDUC2  ...  BPXSY2  BPXDI2  BMXWT  BMXHT  BMXBMI  BMXLEG  \
0       1.0       5.0  ...   124.0    64.0   94.8  184.5    27.8    43.3   
1       2.0       3.0  ...   140.0    88.0   90.4  171.4    30.8    38.0   
2       1.0       3.0  ...   132.0    44.0   83.4  170.1    28.8    35.6   
3       1.0       5.0  ...   134.0    68.0  109.8  160.9    42.4    38.5   
4       1.0       4.0  ...   114.0    54.0   55.2  164.9    20.3    37.4   

   BMXARML  BMXARMC  BMXWAIST  HIQ210  
0     43.6     35.9     101.1     2.0  
1     40.0     33.2     107.9     NaN  
2     37.0     31.0     116.5     2.0  
3     37.7     38.3     110.1     2.0  
4     36.0     27.2      80.4     2.0  

[5 rows x 28 columns]

In [ ]:
mydata.columns

Index(['SEQN', 'ALQ101', 'ALQ110', 'ALQ130', 'SMQ020', 'RIAGENDR', 'RIDAGEYR',
       'RIDRETH1', 'DMDCITZN', 'DMDEDUC2', 'DMDMARTL', 'DMDHHSIZ', 'WTINT2YR',
       'SDMVPSU', 'SDMVSTRA', 'INDFMPIR', 'BPXSY1', 'BPXDI1', 'BPXSY2',
       'BPXDI2', 'BMXWT', 'BMXHT', 'BMXBMI', 'BMXLEG', 'BMXARML', 'BMXARMC',
       'BMXWAIST', 'HIQ210'],
      dtype='object')

In [ ]:
mydata.shape

(5735, 28)

Missing Value

In [ ]:
import numpy as np

mydata.describe(include="all")

SEQN       ALQ101       ALQ110       ALQ130       SMQ020  \
count   5735.000000  5208.000000  1731.000000  3379.000000  5735.000000   
mean   88678.583435     1.336406     1.587522     3.911512     1.607149   
std     2882.139237     0.505514     0.623940    34.341839     0.571975   
min    83732.000000     1.000000     1.000000     1.000000     1.000000   
25%    86164.000000     1.000000     1.000000     1.000000     1.000000   
50%    88668.000000     1.000000     2.000000     2.000000     2.000000   
75%    91178.500000     2.000000     2.000000     3.000000     2.000000   
max    93702.000000     9.000000     9.000000   999.000000     9.000000   

          RIAGENDR     RIDAGEYR     RIDRETH1     DMDCITZN     DMDEDUC2  \
count  5735.000000  5735.000000  5735.000000  5734.000000  5474.000000   
mean      1.518919    48.052310     3.042371     1.185385     3.441725   
std       0.499686    18.431011     1.296793     0.491678     1.309700   
min       1.000000    18.000000     1.000000     1.000000     1.000000   
25%       1.000000    32.000000     2.000000     1.000000     3.000000   
50%       2.000000    48.000000     3.000000     1.000000     4.000000   
75%       2.000000    63.000000     4.000000     1.000000     4.750000   
max       2.000000    80.000000     5.000000     9.000000     9.000000   

          DMDMARTL     DMDHHSIZ       WTINT2YR      SDMVPSU     SDMVSTRA  \
count  5474.000000  5735.000000    5735.000000  5735.000000  5735.000000   
mean      2.628608     3.323801   40312.412352     1.487881   126.236617   
std       2.366786     1.724670   38768.921774     0.499897     4.244406   
min       1.000000     1.000000    5330.960000     1.000000   119.000000   
25%       1.000000     2.000000   17164.085000     1.000000   123.000000   
50%       1.000000     3.000000   24654.860000     1.000000   126.000000   
75%       5.000000     5.000000   42862.305000     2.000000   130.000000   
max      77.000000     7.000000  233755.840000     2.000000   133.000000   

          INDFMPIR       BPXSY1       BPXDI1       BPXSY2       BPXDI2  \
count  5134.000000  5401.000000  5401.000000  5535.000000  5535.000000   
mean      2.403204   125.084614    69.516386   124.783017    69.346703   
std       1.601995    18.480873    12.881575    18.527012    13.022829   
min       0.000000    82.000000     0.000000    84.000000     0.000000   
25%       1.060000   112.000000    62.000000   112.000000    62.000000   
50%       1.980000   122.000000    70.000000   122.000000    70.000000   
75%       3.740000   134.000000    78.000000   134.000000    78.000000   
max       5.000000   236.000000   120.000000   238.000000   144.000000   

             BMXWT        BMXHT       BMXBMI       BMXLEG      BMXARML  \
count  5666.000000  5673.000000  5662.000000  5345.000000  5427.000000   
mean     81.342676   166.142834    29.382197    38.576782    37.146987   
std      21.764409    10.079264     7.095921     3.873018     2.800784   
min      32.400000   129.700000    14.500000    26.000000    28.200000   
25%      65.900000   158.700000    24.300000    36.000000    35.200000   
50%      78.200000   166.000000    28.300000    38.600000    37.100000   
75%      92.700000   173.500000    33.000000    41.200000    39.000000   
max     198.900000   202.700000    67.300000    51.500000    47.400000   

           BMXARMC     BMXWAIST       HIQ210  
count  5427.000000  5368.000000  4732.000000  
mean     33.112235    99.567213     1.915469  
std       5.268027    16.844109     0.414845  
min      17.100000    58.700000     1.000000  
25%      29.500000    87.600000     2.000000  
50%      32.700000    98.300000     2.000000  
75%      36.200000   109.300000     2.000000  
max      58.400000   171.600000     9.000000

In [ ]:
mydata.isnull().sum()

SEQN           0
ALQ101       527
ALQ110      4004
ALQ130      2356
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       1
DMDEDUC2     261
DMDMARTL     261
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
#displaying all the columns

pd.set_option('display.max_columns', None)

mydata.head()

SEQN  ALQ101  ALQ110  ALQ130  SMQ020  RIAGENDR  RIDAGEYR  RIDRETH1  \
0  83732     1.0     NaN     1.0       1         1        62         3   
1  83733     1.0     NaN     6.0       1         1        53         3   
2  83734     1.0     NaN     NaN       1         1        78         3   
3  83735     2.0     1.0     1.0       2         2        56         3   
4  83736     2.0     1.0     1.0       2         2        42         4   

   DMDCITZN  DMDEDUC2  DMDMARTL  DMDHHSIZ   WTINT2YR  SDMVPSU  SDMVSTRA  \
0       1.0       5.0       1.0         2  134671.37        1       125   
1       2.0       3.0       3.0         1   24328.56        1       125   
2       1.0       3.0       1.0         2   12400.01        1       131   
3       1.0       5.0       6.0         1  102718.00        1       131   
4       1.0       4.0       3.0         5   17627.67        2       126   

   INDFMPIR  BPXSY1  BPXDI1  BPXSY2  BPXDI2  BMXWT  BMXHT  BMXBMI  BMXLEG  \
0      4.39   128.0    70.0   124.0    64.0   94.8  184.5    27.8    43.3   
1      1.32   146.0    88.0   140.0    88.0   90.4  171.4    30.8    38.0   
2      1.51   138.0    46.0   132.0    44.0   83.4  170.1    28.8    35.6   
3      5.00   132.0    72.0   134.0    68.0  109.8  160.9    42.4    38.5   
4      1.23   100.0    70.0   114.0    54.0   55.2  164.9    20.3    37.4   

   BMXARML  BMXARMC  BMXWAIST  HIQ210  
0     43.6     35.9     101.1     2.0  
1     40.0     33.2     107.9     NaN  
2     37.0     31.0     116.5     2.0  
3     37.7     38.3     110.1     2.0  
4     36.0     27.2      80.4     2.0

In [ ]:
mydata.isnull()

SEQN  ALQ101  ALQ110  ALQ130  SMQ020  RIAGENDR  RIDAGEYR  RIDRETH1  \
0     False   False    True   False   False     False     False     False   
1     False   False    True   False   False     False     False     False   
2     False   False    True    True   False     False     False     False   
3     False   False   False   False   False     False     False     False   
4     False   False   False   False   False     False     False     False   
...     ...     ...     ...     ...     ...       ...       ...       ...   
5730  False   False   False    True   False     False     False     False   
5731  False   False   False    True   False     False     False     False   
5732  False   False    True   False   False     False     False     False   
5733  False    True    True    True   False     False     False     False   
5734  False   False    True   False   False     False     False     False   

      DMDCITZN  DMDEDUC2  DMDMARTL  DMDHHSIZ  WTINT2YR  SDMVPSU  SDMVSTRA  \
0        False     False     False     False     False    False     False   
1        False     False     False     False     False    False     False   
2        False     False     False     False     False    False     False   
3        False     False     False     False     False    False     False   
4        False     False     False     False     False    False     False   
...        ...       ...       ...       ...       ...      ...       ...   
5730     False     False     False     False     False    False     False   
5731     False     False     False     False     False    False     False   
5732     False     False     False     False     False    False     False   
5733     False     False     False     False     False    False     False   
5734     False     False     False     False     False    False     False   

      INDFMPIR  BPXSY1  BPXDI1  BPXSY2  BPXDI2  BMXWT  BMXHT  BMXBMI  BMXLEG  \
0        False   False   False   False   False  False  False   False   False   
1        False   False   False   False   False  False  False   False   False   
2        False   False   False   False   False  False  False   False   False   
3        False   False   False   False   False  False  False   False   False   
4        False   False   False   False   False  False  False   False   False   
...        ...     ...     ...     ...     ...    ...    ...     ...     ...   
5730     False   False   False   False   False  False  False   False   False   
5731     False   False   False   False   False  False  False   False   False   
5732     False   False   False   False   False  False  False   False   False   
5733     False   False   False   False   False  False  False   False   False   
5734     False   False   False   False   False  False  False   False   False   

      BMXARML  BMXARMC  BMXWAIST  HIQ210  
0       False    False     False   False  
1       False    False     False    True  
2       False    False     False   False  
3       False    False     False   False  
4       False    False     False   False  
...       ...      ...       ...     ...  
5730    False    False     False   False  
5731    False    False     False   False  
5732    False    False      True   False  
5733    False    False     False   False  
5734    False    False     False   False  

[5735 rows x 28 columns]

In [ ]:
mydata.isnull().sum()

SEQN           0
ALQ101       527
ALQ110      4004
ALQ130      2356
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       1
DMDEDUC2     261
DMDMARTL     261
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
#replace the '?' by NaN numpy

import numpy as np

mydata.replace('?', np.nan, inplace = True)

mydata.isnull().sum()

SEQN           0
ALQ101         0
ALQ110      4004
ALQ130      2356
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       1
DMDEDUC2     261
DMDMARTL     261
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
mydata.head()

SEQN  ALQ101  ALQ110  ALQ130  SMQ020  RIAGENDR  RIDAGEYR  RIDRETH1  \
0  83732     1.0     NaN     1.0       1         1        62         3   
1  83733     1.0     NaN     6.0       1         1        53         3   
2  83734     1.0     NaN     NaN       1         1        78         3   
3  83735     2.0     1.0     1.0       2         2        56         3   
4  83736     2.0     1.0     1.0       2         2        42         4   

   DMDCITZN  DMDEDUC2  DMDMARTL  DMDHHSIZ   WTINT2YR  SDMVPSU  SDMVSTRA  \
0       1.0       5.0       1.0         2  134671.37        1       125   
1       2.0       3.0       3.0         1   24328.56        1       125   
2       1.0       3.0       1.0         2   12400.01        1       131   
3       1.0       5.0       6.0         1  102718.00        1       131   
4       1.0       4.0       3.0         5   17627.67        2       126   

   INDFMPIR  BPXSY1  BPXDI1  BPXSY2  BPXDI2  BMXWT  BMXHT  BMXBMI  BMXLEG  \
0      4.39   128.0    70.0   124.0    64.0   94.8  184.5    27.8    43.3   
1      1.32   146.0    88.0   140.0    88.0   90.4  171.4    30.8    38.0   
2      1.51   138.0    46.0   132.0    44.0   83.4  170.1    28.8    35.6   
3      5.00   132.0    72.0   134.0    68.0  109.8  160.9    42.4    38.5   
4      1.23   100.0    70.0   114.0    54.0   55.2  164.9    20.3    37.4   

   BMXARML  BMXARMC  BMXWAIST  HIQ210  
0     43.6     35.9     101.1     2.0  
1     40.0     33.2     107.9     NaN  
2     37.0     31.0     116.5     2.0  
3     37.7     38.3     110.1     2.0  
4     36.0     27.2      80.4     2.0

In [ ]:
mydata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5735 entries, 0 to 5734
Data columns (total 28 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   SEQN      5735 non-null   int64  
 1   ALQ101    5208 non-null   float64
 2   ALQ110    1731 non-null   float64
 3   ALQ130    3379 non-null   float64
 4   SMQ020    5735 non-null   int64  
 5   RIAGENDR  5735 non-null   int64  
 6   RIDAGEYR  5735 non-null   int64  
 7   RIDRETH1  5735 non-null   int64  
 8   DMDCITZN  5734 non-null   float64
 9   DMDEDUC2  5474 non-null   float64
 10  DMDMARTL  5474 non-null   float64
 11  DMDHHSIZ  5735 non-null   int64  
 12  WTINT2YR  5735 non-null   float64
 13  SDMVPSU   5735 non-null   int64  
 14  SDMVSTRA  5735 non-null   int64  
 15  INDFMPIR  5134 non-null   float64
 16  BPXSY1    5401 non-null   float64
 17  BPXDI1    5401 non-null   float64
 18  BPXSY2    5535 non-null   float64
 19  BPXDI2    5535 non-null   float64
 20  BMXWT     5666 non-null   floa

In [ ]:
mydata.describe()

SEQN       ALQ101       ALQ110       ALQ130       SMQ020  \
count   5735.000000  5735.000000  1731.000000  3379.000000  5735.000000   
mean   88678.583435     1.336406     1.587522     3.911512     1.607149   
std     2882.139237     0.481723     0.623940    34.341839     0.571975   
min    83732.000000     1.000000     1.000000     1.000000     1.000000   
25%    86164.000000     1.000000     1.000000     1.000000     1.000000   
50%    88668.000000     1.000000     2.000000     2.000000     2.000000   
75%    91178.500000     2.000000     2.000000     3.000000     2.000000   
max    93702.000000     9.000000     9.000000   999.000000     9.000000   

          RIAGENDR     RIDAGEYR     RIDRETH1     DMDCITZN     DMDEDUC2  \
count  5735.000000  5735.000000  5735.000000  5734.000000  5474.000000   
mean      1.518919    48.052310     3.042371     1.185385     3.441725   
std       0.499686    18.431011     1.296793     0.491678     1.309700   
min       1.000000    18.000000     1.000000     1.000000     1.000000   
25%       1.000000    32.000000     2.000000     1.000000     3.000000   
50%       2.000000    48.000000     3.000000     1.000000     4.000000   
75%       2.000000    63.000000     4.000000     1.000000     4.750000   
max       2.000000    80.000000     5.000000     9.000000     9.000000   

          DMDMARTL     DMDHHSIZ       WTINT2YR      SDMVPSU     SDMVSTRA  \
count  5474.000000  5735.000000    5735.000000  5735.000000  5735.000000   
mean      2.628608     3.323801   40312.412352     1.487881   126.236617   
std       2.366786     1.724670   38768.921774     0.499897     4.244406   
min       1.000000     1.000000    5330.960000     1.000000   119.000000   
25%       1.000000     2.000000   17164.085000     1.000000   123.000000   
50%       1.000000     3.000000   24654.860000     1.000000   126.000000   
75%       5.000000     5.000000   42862.305000     2.000000   130.000000   
max      77.000000     7.000000  233755.840000     2.000000   133.000000   

          INDFMPIR       BPXSY1       BPXDI1       BPXSY2       BPXDI2  \
count  5134.000000  5401.000000  5401.000000  5535.000000  5535.000000   
mean      2.403204   125.084614    69.516386   124.783017    69.346703   
std       1.601995    18.480873    12.881575    18.527012    13.022829   
min       0.000000    82.000000     0.000000    84.000000     0.000000   
25%       1.060000   112.000000    62.000000   112.000000    62.000000   
50%       1.980000   122.000000    70.000000   122.000000    70.000000   
75%       3.740000   134.000000    78.000000   134.000000    78.000000   
max       5.000000   236.000000   120.000000   238.000000   144.000000   

             BMXWT        BMXHT       BMXBMI       BMXLEG      BMXARML  \
count  5666.000000  5673.000000  5662.000000  5345.000000  5427.000000   
mean     81.342676   166.142834    29.382197    38.576782    37.146987   
std      21.764409    10.079264     7.095921     3.873018     2.800784   
min      32.400000   129.700000    14.500000    26.000000    28.200000   
25%      65.900000   158.700000    24.300000    36.000000    35.200000   
50%      78.200000   166.000000    28.300000    38.600000    37.100000   
75%      92.700000   173.500000    33.000000    41.200000    39.000000   
max     198.900000   202.700000    67.300000    51.500000    47.400000   

           BMXARMC     BMXWAIST       HIQ210  
count  5427.000000  5368.000000  4732.000000  
mean     33.112235    99.567213     1.915469  
std       5.268027    16.844109     0.414845  
min      17.100000    58.700000     1.000000  
25%      29.500000    87.600000     2.000000  
50%      32.700000    98.300000     2.000000  
75%      36.200000   109.300000     2.000000  
max      58.400000   171.600000     9.000000

Column with missing values:
ALQ101       527
ALQ110      4004
ALQ130      2356
DMDCITZN       1
DMDEDUC2     261
DMDMARTL     261
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003

In [ ]:
mean_ALQ101 = mydata['ALQ101'].astype('float').mean()

mean_ALQ101

1.336405529953917

In [ ]:
mydata['ALQ101'].replace(np.nan, mean_ALQ101, inplace=True)

mydata.isnull().sum()

SEQN           0
ALQ101         0
ALQ110         0
ALQ130      2356
SMQ020         0
RIAGENDR       0
RIDAGEYR       0
RIDRETH1       0
DMDCITZN       1
DMDEDUC2     261
DMDMARTL     261
DMDHHSIZ       0
WTINT2YR       0
SDMVPSU        0
SDMVSTRA       0
INDFMPIR     601
BPXSY1       334
BPXDI1       334
BPXSY2       200
BPXDI2       200
BMXWT         69
BMXHT         62
BMXBMI        73
BMXLEG       390
BMXARML      308
BMXARMC      308
BMXWAIST     367
HIQ210      1003
dtype: int64

In [ ]:
mydata['ALQ101'].value_counts()

ALQ101
1.000000    3477
2.000000    1728
1.336406     527
9.000000       3
Name: count, dtype: int64